### PR TITLE
Object pose estimation and also some pipeline enhancement 

### LINKED ISSUE #2284: Add object pose estimation
For things like balls which are the same from all sides this is easy we just throw it threw solvePNP. For other items we can use an OBB + side ratio to figure out what face we are looking at and then pipe through solvePNP I think?

### PR DESCRIPTION
**Pull request by Ruthie-FRC**
## Description

What changed? Why? (the code + comments should speak for itself on the "how")

Include fun testing screenshots or a cool video, to collect test evidence in a place where we can later reference it. Including proof this change was tested makes reviewing easier, helps us make sure we tested all our edge cases, and helps provide context for the future.

Any issues this pull request closes or pull requests this supersedes should be linked with `Closes #issuenumber`.

improved object detection pipeline configuration and processing.  enhancements to 3D processing in the object detection pipeline, adjustments to default settings, and improvements to configuration options filtering.

closes #2284 

## Meta

Merge checklist:
- [ ] Pull Request title is [short, imperative summary](https://cbea.ms/git-commit/) of proposed changes
- [ ] The description documents the _what_ and _why_, including events that led to this PR
- [ ] If this PR changes behavior or adds a feature, user documentation is updated
- [ ] If this PR touches photon-serde, all messages have been regenerated and hashes have not changed unexpectedly
- [ ] If this PR touches configuration, this is backwards compatible with all settings going back to the previous seasons's last release (seasons end after champs ends)
- [ ] If this PR touches pipeline settings or anything related to data exchange, the frontend typing is updated
- [ ] If this PR addresses a bug, a regression test for it is added
- [ ] If this PR adds a dependency, the license has been checked for compatibility and steps taken to follow it

### CODE REVIEW

**Comment by Ruthie-FRC** [comment]
this is just a first attempt at it but figured I might as well get some feedback

File: photon-client/src/types/PipelineTypes.ts [hunk 1]
```diff
@@ -30,9 +30,9 @@ export enum TargetModel {
   InfiniteRechargeHighGoalOuter = 2,
   CircularPowerCell7in = 3,
   RapidReactCircularCargoBall = 4,
-  AprilTag6in_16h5 = 5,
```
Comment (gerth2) [inline photon-client/src/types/PipelineTypes.ts:33]: Why are these reordered?

File: photon-client/src/types/PipelineTypes.ts [hunk 2]
```diff
@@ -313,8 +313,8 @@ export const DefaultObjectDetectionPipelineSettings: ObjectDetectionPipelineSett
   ...DefaultPipelineSettings,
   pipelineType: PipelineType.ObjectDetection,
   cameraGain: 20,
-  targetModel: TargetModel.InfiniteRechargeHighGoalOuter,
-  ledMode: true,
+  targetModel: TargetModel.ReefscapeAlgae,
```
Comment (gerth2) [inline photon-client/src/types/PipelineTypes.ts:316]: Why is this changed?

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/ObjectDetectionPipe.java [hunk 3]
```diff
@@ -52,7 +52,7 @@ protected List<NeuralNetworkPipeResult> process(CVMat in) {
             return List.of();
         }
 
-        return detector.detect(in.getMat(), params.nms(), params.confidence());
```
Comment (gerth2) [inline photon-core/src/main/java/org/photonvision/vision/pipe/impl/ObjectDetectionPipe.java:55]: why was this changed? there needs to be great caution with where mat's get returned, we've had memory leak issues from things like this in the past

**Review by gerth2 — COMMENTED** [review]
Any screenshots or videos of this working? No testcases were updated

**Comment by gerth2** [comment]
as a general note, this is a sizeable feature add. It should probably wait for a post-champs merge.

File: photon-core/src/main/java/org/photonvision/vision/pipe/impl/ObjectDetectionPipe.java (hunk 3 above)
Comment (Ruthie-FRC) [reply photon-core/src/main/java/org/photonvision/vision/pipe/impl/ObjectDetectionPipe.java:55]: will revert.

File: photon-client/src/types/PipelineTypes.ts (hunk 2 above)
Comment (Ruthie-FRC) [reply photon-client/src/types/PipelineTypes.ts:316]: that.... that was me being an idiot and forgetting how things work. sorry.

**Comment by Ruthie-FRC** [comment]
> Any screenshots or videos of this working? No testcases were updated
I need to test run it when i get to a proper laptop, i knew it wasnt ready for merge just wanted an idea of what egregious mistakes i should probably fix.

**Comment by Ruthie-FRC** [comment]
i promise i have not abandoned this, nor have i any of my prs. im jsut a bit more busy now with me developing my own physics library. I'll probably get these finshed up after finals

**Comment by Ruthie-FRC** [comment]
ok im going to be honest i am working on my own vendordep now and i owe it to you guys to close this issue and not kidn of barely update such a large change in a pr. for now im oging to close it, i acan alwoasy make anotehr one later. thank you for everhtnig
